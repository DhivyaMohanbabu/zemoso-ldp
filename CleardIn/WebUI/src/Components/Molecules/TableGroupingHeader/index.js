/* eslint-disable max-len */
/* eslint-disable react/prop-types */
/* eslint-disable react/no-typos */
/* eslint-disable react/no-unescaped-entities */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from 'prop-types';
import { Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import ColoredOption from '../../Atoms/ColoredOption';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    padding: '0.40rem 0.5rem 0.5rem 0',
    height: 62,
    alignContent: 'center',
  },
  label: {
    direction: 'row',
    color: theme.palette.clearedIn.grey[500],
    fontSize: theme.typography.button.fontSize,
    fontWeight: theme.typography.button.fontWeight,
    flexGrow: 1,
  },
  text: {
    marginLeft: '4px',
  },
  count: {
    margin: '24px 32px 17px',
    textAlign: 'right',
    color: theme.palette.clearedIn.grey[500],
    fontWeight: theme.typography.button.fontWeight,
  },
}));

function TableGroupingHeader({ text, count }) {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <div className={classes.label}>
        <label className={classes.text}>Group</label>
        <ColoredOption label={text} />
      </div>
      <Typography className={classes.count} data-testid="count">
        {`Count ${count}`}
      </Typography>
    </div>
  );
}
TableGroupingHeader.propTypes = {
  text: PropTypes.string.isRequired,
  count: PropTypes.string.isRequired,
};
export default TableGroupingHeader;
