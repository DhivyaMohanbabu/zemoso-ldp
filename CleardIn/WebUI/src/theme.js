import { grey } from '@material-ui/core/colors';
import { createMuiTheme } from '@material-ui/core/styles';
// import 'typeface-roboto';

const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#1665d8',
      light: '#b0f1f0',
    },
    secondary: {
      main: '#9ea0a5',
      dark: grey,
    },
    error: {
      main: '#ffada5',

    },
    success: {
      main: '#daefa9',
    },
    info: {
      main: grey[900],
      light: '#fdd570',
      dark: grey[600],
    },
    text: {
      primary: '#1c1c21', // '#3e3f42',
      secondary: '#5c5c5c', // #6b6c6f',
    },
    grey: {
      100: '#f8fafb',
      200: '#f2f4f6',
      300: '#e8ecef',
      400: '#98a9bc',
      500: '#778ca2',
    },
    clearedIn: {
      grey: {
        100: '#f6f6f6',
        200: '#eeeeee',
        300: '#cccccc',
        400: '#949494',
        500: '#666666',
      },
      selectedElementBGColor: '#eaf0ff', // rgba(77, 124, 254, 0.12),
      dark: '#5c5c5c',
      bodyBackground: '#fafafa',
      blue: '#1d4cd7',
    },
    relationShip: {
      vendor: '#daefa9',
      customer: '#fdd570',
      partner: '#b0f1f0',
    },
  },
  typography: {
    fontFamily: ['Roboto', 'Rubik', 'sans-serif'].join(','),
    h1: {
      fontSize: '1.875rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h2: {
      fontSize: '1.625rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h3: {
      fontSize: '1.25rem',
      fontWeight: 300,
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h4: {
      fontSize: '1.125rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h5: {
      fontSize: '1rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 1.5,
      letterSpacing: 'normal',
    },
    h6: {
      fontSize: '1.25rem',
      fontWeight: 500,
      fontStyle: 'normal',
      lineHeight: 1.05,
      letterSpacing: 'normal',
    },
    subtitle1: {
      fontSize: '0.938rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    subtitle2: {
      fontSize: '0.938rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    caption: {
      fontSize: '0.875rem',
      fontWeight: 500,
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    link: {
      fontSize: '0.875rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 1.5,
      letterSpacing: 'normal',
    },
    body1: {
      fontSize: '0.875rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 1.14,
      letterSpacing: 'normal',
    },
    body2: {
      fontFamily: 'Rubik',
      fontSize: '0.875rem',
      fontWeight: 300,
      fontStyle: 'normal',
      lineHeight: 1.5,
      letterSpacing: 'normal',
    },
    button: {
      fontSize: '0.75rem',
      fontWeight: 500,
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
      textTransform: 'none',
    },
  },
  overrides: {
    MuiButton: {
      root: {
        maxWidth: '200px',
        display: 'block',
      },
      label: {
        whiteSpace: 'nowrap',
        overflow: 'hidden',
        textOverflow: 'ellipsis',
      },
    },
  },
});

export default theme;
