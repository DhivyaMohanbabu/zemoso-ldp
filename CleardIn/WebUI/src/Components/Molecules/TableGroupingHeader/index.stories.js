import React from 'react';
import TableGroupingHeader from '.';

export default {
  title: 'Molecule/TableGroupingHeader',
};

const Template = (args) => <TableGroupingHeader {... args} />;

export const Default = Template.bind({});
Default.args = {
  count: 5,
  text: 'Do not trust',
};
