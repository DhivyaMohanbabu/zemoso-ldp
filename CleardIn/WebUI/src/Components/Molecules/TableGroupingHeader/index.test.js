import React from 'react';
import { ThemeProvider } from '@material-ui/core/styles';
import { render } from '@testing-library/react';
import TableGroupingHeader from '.';
import '@testing-library/jest-dom';
import theme from '../../../theme';

describe('TableHeader', () => {
  it('match tableHeader snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <TableGroupingHeader text="Do not trust" count="Count 5" />
      </ThemeProvider>,
    );
    expect(component).toMatchSnapshot();
  });
  it('should render count of rows in table', () => {
    const { queryByTestId, getByText } = render(
      <ThemeProvider theme={theme}>
        <TableGroupingHeader text="Do not trust" count="3" />
      </ThemeProvider>,
    );
    const component = queryByTestId('count');
    expect(component).not.toBeNull();
    expect(getByText('Count 3')).toBeVisible();
  });
});
