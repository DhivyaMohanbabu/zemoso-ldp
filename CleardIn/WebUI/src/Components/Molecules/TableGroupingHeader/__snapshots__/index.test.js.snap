// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader match tableHeader snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
      >
        <div
          class="makeStyles-label-2"
        >
          <label
            class="makeStyles-text-3"
          >
            Group
          </label>
          <div>
            <div
              class="MuiChip-root makeStyles-donottrust-7 makeStyles-chipFont-9"
              data-testid="chip"
            >
              <span
                class="MuiChip-label"
              >
                Do not trust
              </span>
            </div>
          </div>
        </div>
        <p
          class="MuiTypography-root makeStyles-count-4 MuiTypography-body1"
          data-testid="count"
        >
          Count Count 5
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
    >
      <div
        class="makeStyles-label-2"
      >
        <label
          class="makeStyles-text-3"
        >
          Group
        </label>
        <div>
          <div
            class="MuiChip-root makeStyles-donottrust-7 makeStyles-chipFont-9"
            data-testid="chip"
          >
            <span
              class="MuiChip-label"
            >
              Do not trust
            </span>
          </div>
        </div>
      </div>
      <p
        class="MuiTypography-root makeStyles-count-4 MuiTypography-body1"
        data-testid="count"
      >
        Count Count 5
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
